api/src/Entity/ContentNode.php

```diff
@@ -41,7 +42,8 @@
     denormalizationContext: ['groups' => ['write']],
     normalizationContext: ['groups' => ['read']],
 )]
-#[ApiFilter(SearchFilter::class, properties: ['parent', 'contentType', 'root'])]
+#[ApiFilter(SearchFilter::class, properties: ['contentType', 'root'])]
```

Comment:
Why remove the `parent` filter? Are you sure that it wasn't used anywhere? I can't remember, but there were a lot of tests for this which you had to remove.

Reply:
Removing the `parent` filter results in children property to not be a single link but an array of links (basically disabling RelatedCollectionLinkNormalizer). This avoids a lot of above API Calls, because once I have the complete list of ContentNodes I can freely traverse down the child nodes without additional API calls.

`hal-json-vuex` is now "immune" to this, because it will just create a virtual key for the children array.

It shouldn't break anything. Just notice that in the frontend, we're actually no using the children property at the moment, because we do the filtering manually:
https://github.com/ecamp/ecamp3/blob/devel/frontend/src/components/activity/DraggableContentNodes.vue#L67

Reply:
Ah, now I get it. In this specific case we actually prefer the array of hrefs `children: [{ href: '/content-nodes/1a2b3c4d' }, { href: '/content-nodes/1234abcd' }, ... ]` over the shortened, filtered single link `children: { href: '/content-nodes?parent=/content-nodes/00000000' }`, because we don't want to embed all children, and because the array contains all the information we need for filtering in the frontend, without sending a separate request for /content-nodes?parent=/content-nodes/00000000

When I wrote the RelatedCollectionLinkNormalizer, I did not foresee that the link array could ever be more useful than the single filtered link. Maybe we don't need the parent filter right now. But I think we should introduce a way to explicitly disable the filtered link even when the filter exists. Otherwise, someone might in the future e.g. add filters for all relations, and make the performance worse without noticing.

Reply:
TODO: Explicitly comment in tests why children needs to be an array (to avoid someone adding the parent filter again in the future)